### DISCUSSION TITLE
Upstream changes

### DISCUSSION THREAD

**Discussion by MariusGoehring** [kimai/kimai#4227] [General]
Dear Kimai Team,

we here at Helsinki Systems would like to suggest some upstream changes.
First is a Activity User Budget. That is, you can assign a budget to a User for a specific activity.
Second is a flexible time period for the reporting views. That is, you can select a start date and a end date for the reporting view. It would be for the project details report and the user reports.

would be happy to hear from you, what you think about this. If you agree we would develop these features.

Best regards 
Marius

**Comment by kevinpapst** [comment]
Hi Marius,

1. Not a core feature
2. First reaction: yes. Second reaction: many questions, e.g. 
  - Does it make any sense for the project detail report? I doubt it. 
  - What about the different user reports? At least year looks different. How do you display 3,5 months? 

Not really sure if any of that is a good idea. 
I am not saying no, more like: convince me with more input.

**Reply by MariusGoehring** [reply]
Hi Kevin, 
thanks for the reply. For some users/companies it might make sense to have a flexible reporting period. So you can exactly see in which time period how much the colleagues worked and how much revenue was gained. Regarding the second question, it would be monthwise, so frpom a start month to an end month, my bad in terms of expression .... 
So we would adapt the yearly user report to a flexible time period with monthly overview, I guess this has the same benefits as for the project detail: You are not limited to viewing 'only one year' of statistics... I hope this makes sense.
